src/Database/LSMTree/Internal/BitVec.hs

```diff
@@ -0,0 +1,443 @@
+{-# LANGUAGE CPP                        #-}
+
+{-# LANGUAGE BangPatterns               #-}
+{-# LANGUAGE DeriveDataTypeable         #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase                 #-}
+{-# LANGUAGE MagicHash                  #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE UnboxedTuples              #-}
+{-# LANGUAGE ViewPatterns               #-}
+
+-- | This is Data.Bit.Internal module from @bitvec@ package,
+-- however explicitly using 'Word64', and not machine native word.
```

Comment:
Let's also specify somewhere in this module that the `Vector Bit` doesn't guarantee thread safety